package/network/services/dnsmasq/files/dnsmasq.init

```diff
@@ -289,7 +290,7 @@ dhcp_host_add() {
 
 	config_get leasetime "$cfg" leasetime
 
-	xappend "--dhcp-host=$macs${networkid:+,net:$networkid}${broadcast:+,set:needs-broadcast}${tag:+,set:$tag}${ip:+,$ip${hostid:+,[::$hostid]}}${name:+,$name}${leasetime:+,$leasetime}"
+	xappend "--dhcp-host=$macs${duid:+,id:$duid}${networkid:+,net:$networkid}${broadcast:+,set:needs-broadcast}${tag:+,set:$tag}${ip:+,$ip${hostid:+,[::$hostid]}}${name:+,$name}${leasetime:+,$leasetime}"
```

Comment:
There must be separate uci options for the DHCPv6 'duid' and DHCPv4 'clientid' because they are not the same, even dnsmasq (wrongly) treats them the same.
In a default LEDE/OpenWrt installation, DHCPv4 is handled by dnsmasq and DHCPv6 by odhcpd. So 'clientid' must in that case only be used by dnsmasq and 'duid' must end up only in the odhcpd configuration as they are not the same thing.

I'm not sure what to do in case dnsmasq is used for both DHCPv4 and DHCPv6. Maybe in that case we should opt for two dhcp-host lines, one for the DHCPv4 address using the 'clientid' and add another line for the DHCPv6 address with the 'duid'. I didn't check if dnsmasq allows listing the id: twice on the same dhcp-host line, in which case the solution would be trivial.